src/Microsoft.OData.Core/Evaluation/ODataMetadataContext.cs

```diff
@@ -95,7 +95,7 @@ internal sealed class ODataMetadataContext : IODataMetadataContext
         /// <summary>
         /// The OData Uri.
         /// </summary>
-        private readonly ODataUri odataUri;
+        private readonly ODataUriSlim? odataUri;

```

Comment:
Because in the previous code, an `ODataMetadataContext` can be created with a `null` `ODataUri`. This happens in [ODataJsonLightDeserializer(https://github.com/OData/odata.net/blob/master/src/Microsoft.OData.Core/JsonLight/ODataJsonLightDeserializer.cs#L126)]. The deserializer passes the `this.ODataUri` property to the `ODataMetadataContext` constructor. The `ODataJsonLightDeserializer.ODataUri` property can be null as seen [here](https://github.com/OData/odata.net/blob/master/src/Microsoft.OData.Core/JsonLight/ODataJsonLightDeserializer.cs#L106).

So for this PR, I needed a way to handle the situation where `IODataMetadataContext.ODataUri` can be null. Since `ODataUriSlim` is a struct, it can't be null, so that's why I decided to use the nullable version `ODataUriSlim?` for that. `Nullable<T>` is a struct, so we don't incur extra heap allocations when making the struct nullable.

Reply:
@KenitoInc I'm guessing because when `ODataUri` class was used in the field declaration, it could accommodate nulls out of the box. It's therefore necessary to declare the field as nullable when using `ODataUriSlim` struct (value type)